src/parser/constructs.rs

```diff
@@ -273,12 +244,18 @@ impl Construct {
     /// `<arg_list> := [(<constant> | <variable> | <instruction>)*]`
     /// `<identifier> ( <arg_list> )`
     pub(crate) fn function_call(input: &str) -> ParseResult<&str, FunctionCall> {
-        let call = alt((
-            Construct::function_call_no_args,
-            Construct::function_call_args,
-        ))(input)?;
+        let (input, fn_id) = Token::identifier(input)?;
+        let (input, _) = Token::left_parenthesis(input)?;
```

Comment:
Hey, there's a bug here :D (Which was present in the original impl, no worries)

Could you add a test with the following function call and declaration (in another PR if possible, or I can also fix it but if you don't want to do it I'll do it :) )
```rust
func fn_name () {}
fn_name ()
// Note the extra whitespaces before the parentheses
```